app/controllers/postgresql_lo_streamer/lo_controller.rb

```diff
@@ -28,23 +28,59 @@ def connection
     end
 
     private
-      
+
     def object_exists?(identifier)
       begin
         connection.lo_open(identifier, ::PG::INV_READ)
       rescue PG::Error => e
         if e.to_s.include? "does not exist"
           return false
         end
-          
+
         raise
       end
 
       return true
     end
-      
+
     def configuration
       PostgresqlLoStreamer.configuration
     end
+
+    def headers_for_extension_or_default
+      #extension provided and recognized
+      #Mime is built into rails
+      if params[:format].present? && Mime::Type.lookup_by_extension(params[:format]).present?
+        type = Mime::Type.lookup_by_extension(params[:format]).to_s
+        {type: type, disposition: disposition_from_type(type) }
+      else
+        configuration.options #fallback
+      end
+    end
+
+    def disposition_from_type(type)
+      #subjective switch statement, should be able to configure eventually
+      inline_types = [
+        "image/jpeg",
+        "image/png",
+        "image/gif",
+        "image/svg+xml",
+        "text/css",
+        "text/plain"
+      ]
+      attachment_types = [
+        "text/csv"
+      ]
+
+      case type
+      when *inline_types
+        "inline"
+      when *attachment_types
+        "attachment"
+      else
+        "attachment" #fallback, subjective
```

Comment:
Since attachement is our default I believe it's better to drop the `attachment_types` and specify only the inline types.

Reply:
I agree 😆.. wanted to show you it could be either way